consensus/src/quorum_store/batch_generator.rs

```diff
@@ -404,6 +408,15 @@ impl BatchGenerator {
                         let batches = self.handle_scheduled_pull(dynamic_pull_max_txn).await;
                         if !batches.is_empty() {
                             last_non_empty_pull = tick_start;
+
+                            let persist_start = Instant::now();
+                            let mut persist_requests = vec![];
+                            for batch in batches.clone().into_iter() {
+                                persist_requests.push(batch.into());
+                            }
+                            self.batch_writer.persist(persist_requests);
```

Comment:
Did you observe any bottlenecks here? This might need to be parallelized or sent to another process, or this can block the loop for a while.

Reply:
Running max_load_large I didn't see any issues